.cirrus.yml

```diff
@@ -39,742 +25,235 @@ env:
-        - "vendor"
-        - "varlink_api"
-
-    only_if: >-
-        $CIRRUS_BRANCH != $DEST_BRANCH &&
-        $CIRRUS_CHANGE_MESSAGE !=~ '.*CI:IMG.*' &&
-        $CIRRUS_CHANGE_MESSAGE !=~ '.*CI:DOCS.*'
-
-    gce_instance:
-        cpu: 8
-        memory: "8Gb"
-
-    env:
-        MOD_LIBPOD_CONF: 'false'
-
-    timeout_in: 30m
-
-    setup_environment_script: '$SCRIPT_BASE/setup_environment.sh |& ${TIMESTAMP}'
-    build_without_cgo_script:
-        - 'source $SCRIPT_BASE/lib.sh'
-        - 'make build-no-cgo'
-
-    on_failure:
-        failed_branch_script: '$CIRRUS_WORKING_DIR/$SCRIPT_BASE/notice_branch_failure.sh'
-
-
-# Update metadata on VM images referenced by this repository state
-meta_task:
-
-    depends_on:
-        - "gating"
-        - "vendor"
-        - "varlink_api"
-        - "build_each_commit"
-        - "build_without_cgo"
-
-    container:
-        image: "quay.io/libpod/imgts:master"  # see contrib/imgts
-        cpu: 1
-        memory: 1
-
-    env: &meta_env_vars
-        # Space-separated list of images used by this repository state
-        IMGNAMES: >-
-            ${FEDORA_CACHE_IMAGE_NAME}
-            ${PRIOR_FEDORA_CACHE_IMAGE_NAME}
-            ${UBUNTU_CACHE_IMAGE_NAME}
-            ${PRIOR_UBUNTU_CACHE_IMAGE_NAME}
-            ${IMAGE_BUILDER_CACHE_IMAGE_NAME}
-        BUILDID: "${CIRRUS_BUILD_ID}"
-        REPOREF: "${CIRRUS_CHANGE_IN_REPO}"
-        GCPJSON: ENCRYPTED[950d9c64ad78f7b1f0c7e499b42dc058d2b23aa67e38b315e68f557f2aba0bf83068d4734f7b1e1bdd22deabe99629df]
-        # needed for output-masking purposes
-        GCPNAME: ENCRYPTED[b05d469a0dba8cb479cb00cc7c1f6747c91d17622fba260a986b976aa6c817d4077eacffd4613d6d5f23afc4084fab1d]
-        GCPPROJECT: ENCRYPTED[7c80e728e046b1c76147afd156a32c1c57d4a1ac1eab93b7e68e718c61ca8564fc61fef815952b8ae0a64e7034b8fe4f]
-
-    timeout_in: 10m
-
-    # Cirrus-CI ignores entrypoint defined in image
-    script: '$CIRRUS_WORKING_DIR/$SCRIPT_BASE/update_meta.sh |& ${TIMESTAMP}'
-
-
-# Remove old and disused images based on labels set by meta_task
-image_prune_task:
-
-    # This should ONLY ever run from the master branch, and never
-    # anywhere else so it's behavior is always consistent, even
-    # as new branches are created.
-    only_if: $CIRRUS_BRANCH == "master"
-
-    depends_on:
-        - "meta"
-
-    container:
-        image: "quay.io/libpod/imgprune:master"  # see contrib/imgprune
-        cpu: 1
-        memory: 1
-
-    env:
-        <<: *meta_env_vars
-        GCPJSON: ENCRYPTED[4c11d8e09c904c30fc70eecb95c73dec0ddf19976f9b981a0f80f3f6599e8f990bcef93c253ac0277f200850d98528e7]
-        GCPNAME: ENCRYPTED[7f54557ba6e5a437f11283a53e71baec9ca546f48a9835538cc54d297f79968eb1337d4596a1025b14f9d1c5723fbd29]
-
-    timeout_in: 10m
-
-    script: '/usr/local/bin/entrypoint.sh |& ${TIMESTAMP}'
-
+    FEDORA_CONTAINER_FQIN: "quay.io/libpod/in_podman_fedora:${DEST_BRANCH}"
+    UBUNTU_CONTAINER_FQIN: "quay.io/libpod/in_podman_ubuntu:${DEST_BRANCH}"
```

Comment:
`contrib/curris/lib.sh` (and two others) still reference `IN_PODMAN_IMAGE` pointing to `quay.io/libpod/in_podman:` (as in just `in_podman`, not `in_podman_distro`). Do those need to be updated? (Also, how do these `in_podman_xxx` images get created and pushed to quay?)

Reply:
Oh good catch, I was thinking of moving those to a separate follow-on PR (along with relocating the VM Image build stuff).  Now I'm realizing this PR needs to depend on that.  The reason is the answer to your other question:

> how do these in_podman_xxx images get created and pushed to quay?

We adopted the "easy-way" here, and it's outlived its usefulness.  It works like this:

1. A branch (like master) changes due to a PR merge or a tag push
2. On quay.io, we have many images defined, including `in_podman`.  Many have a build-trigger defined for any branch, though `in_podman` and several others have specific triggers on named branches (for historical reasons).
3. One/more of these triggers notice the repository change, and independently fire-off `docker build ...` jobs (within quay.io service).
4. Every build pushes both a `latest and named tag (like `master`, or 'v1.6', etc) depending on the build trigger configuration.
5. Each quay.io repo has notification settings configured to e-mail me when a build fails.